core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -327,10 +327,15 @@ class LogManager(val logDirs: Array[File],
         val needToStopCleaner: Boolean = truncateOffset < log.activeSegment.baseOffset
```

Comment:
We can simplify the code by making this:

```scala
val needToStopCleaner = cleaner != null && truncateOffset < log.activeSegment.baseOffset
```

So subsequent checks only need to check the boolean.